Conda release setup

AjayThorve opened the pull request:
- Added setup files and conda recipes

<!--

Thank you for contributing to cuxfilter :)

Here are some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that you have written units tests for the changes made/features
   added.

3. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

4. If your pull request is not ready for review but you want to make use of the
   continuous integration testing facilities please label it with `[WIP]`.

5. If your pull request is ready to be reviewed without requiring additional
   work on top of it, then remove the `[WIP]` label (if present) and replace
   it with `[REVIEW]`. If assistance is required to complete the functionality,
   for example when the C/C++ code of a feature is complete but Python bindings
   are still required, then add the label `[HELP-REQ]` so that others can triage
   and assist. The additional changes then can be implemented on top of the
   same PR. If the assistance is done by members of the rapidsAI team, then no
   additional actions are required by the creator of the original PR for this,
   otherwise the original author of the PR needs to give permission to the
   person(s) assisting to commit to their personal fork of the project. If that
   doesn't happen then a new PR based on the code of the original PR can be
   opened by the person assisting, which then will be the PR that will be
   merged.

6. Once all work has been done and review has taken place please do not add
   features or make changes out of the scope of those requested by the reviewer
   (doing this just add delays as already reviewed code ends up having to be
   re-reviewed/it is hard to tell what is new etc!). Further, please do not
   rebase your branch on master/force push/rewrite history, doing any of these
   causes the context of any comments made by reviewers to be lost. If
   conflicts occur against master they should be resolved by merging master
   into the branch used for making the pull request.

Many thanks in advance for your cooperation!

-->

dillon-cullinan commented on build.sh:
```diff
@@ -0,0 +1,129 @@
+    cd ${REPODIR}/../
+    git clone https://github.com/rapidsai/cuDataShader.git
+    cd ${REPODIR}/../cuDataShader
+    python setup.py install --single-version-externally-managed --record=record.txt
```
I would `cd` back to `$REPODIR` after this installation. I'm also unsure how this will act in a conda build environment, but we can leave it as is and see.

dillon-cullinan commented on build.sh:
```diff
@@ -0,0 +1,129 @@
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean cuXfilter -v -g -n --allgpuarch -h"
```
All lowercase args

dillon-cullinan commented on build.sh:
```diff
@@ -0,0 +1,129 @@
+   --allgpuarch - build for all supported GPU architectures
+   -h           - print this text
+"
+CUXFILTER_BUILD_DIR=${REPODIR}/python/cuXfilter/build
```
All lower case directories as well. Should change the python folder name to all lower in the repo as well.

dillon-cullinan commented on ci/cpu/build.sh:
```diff
@@ -0,0 +1,63 @@
+################################################################################
+
+logger "Build conda pkg for cuxfilter..."
+source ci/cpu/cuxfilter/build_cuXfilter.sh
```
Lowercase script names as well.

dillon-cullinan commented on ci/cpu/prebuild.sh:
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+#Upload cudf once per PYTHON
```
cudf --> cuxfilter

dillon-cullinan commented on ci/cpu/prebuild.sh:
```diff
@@ -0,0 +1,15 @@
+    export UPLOAD_LIBCUDF=1
+else
+    export UPLOAD_LIBCUDF=0
+fi
```
This block is unneeded as you are python only.

dillon-cullinan commented on ci/gpu/build.sh:
```diff
@@ -0,0 +1,80 @@
+
+    cd $WORKSPACE/python/cuXfilter/tests
+    logger "Python py.test for cuXfilter..."
+    py.test --cache-clear --junitxml=${WORKSPACE}/junit-cuXfilter.xml -v --cov-config=.coveragerc --cov=cuXfilter --cov-report=xml:${WORKSPACE}/python/cuXfilter/cuXfilter-coverage.xml --cov-report term
```
All of the `-cov` arguments should be removed, they require CodeCov integration to do anything. Everything up to the `-v` stays, the rest should be deleted.

dillon-cullinan commented on conda/recipes/cuXfilter/build.sh:
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2018-2019, NVIDIA CORPORATION.
+
+# This assumes the script is executed from the root of the repo directory
+./build.sh cudf
```
Should build cuxfilter and cudatashader here. Also folder name to all lowercase.

dillon-cullinan commented on conda/recipes/cuXfilter/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+    - setuptools
+  run:
+    - python
+    - cudf=0.10
```
```suggestion
    - cudf {{ minor_version }}
```

AjayThorve commented:
Hey Dillon, I have updated the files as per the requested changes. CuDatashader build path is still not all lowercase though, as the repo name itself is not all lower, and I believe I don't have the privileges to update that. Apart from that, I have updated everything else.

exactlyallan commented on conda/environments/cuXfilter_dev_cuda10.0.yml:
```diff
@@ -0,0 +1,33 @@
+- nbsphinx
+- numpydoc
+- ipython
+- altair
```
Do we still need altair?

AjayThorve replied:
will remove it
